src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonService.kt

```diff
@@ -127,36 +134,108 @@ class PrisonService(
+    val oldEmailAddress = contactDetails.emailAddress?.value
     if (oldEmailAddress != newEmailAddress) {
-      val persistedEmailAddress = createOrGetEmailAddress(newEmailAddress)
+      updateEmailAddress(newEmailAddress, contactDetails, oldEmailAddress)
+    }
+    return SetOutcome.UPDATED
+  }
+
+  private fun updateEmailAddress(
+    newEmailAddress: String,
+    contactDetails: ContactDetails,
+    oldEmailAddress: String?,
+  ) {
+    val persistedEmailAddress = createOrGetEmailAddress(newEmailAddress)
 
-      persistedEmailAddress.contactDetails.add(contactDetails)
-      contactDetails.emailAddress = persistedEmailAddress
-      contactDetailsRepository.saveAndFlush(contactDetails)
+    persistedEmailAddress.contactDetails.add(contactDetails)
+    contactDetails.emailAddress = persistedEmailAddress
+    contactDetailsRepository.saveAndFlush(contactDetails)
 
+    oldEmailAddress?.let {
       if (contactDetailsRepository.isEmailOrphaned(oldEmailAddress)) {
         emailAddressRepository.delete(oldEmailAddress)
       }
     }
+  }
+
+  @Transactional
+  @PreAuthorize(CLIENT_CAN_MAINTAIN_ADDRESSES)
+  fun setTelephoneAddress(prisonId: String, newTelephoneAddress: String, departmentType: DepartmentType): SetOutcome {
+    val contactDetails = contactDetailsRepository.getByPrisonIdAndType(prisonId, departmentType)
+    if (contactDetails == null) {
+      val prison = prisonRepository.getReferenceById(prisonId) ?: throw EntityNotFoundException()
+      val persistedTelephoneAddress = createOrGetTelephoneAddress(newTelephoneAddress)
+      contactDetailsRepository.saveAndFlush(ContactDetails(prisonId, prison, departmentType, telephoneAddress = persistedTelephoneAddress))
+      return SetOutcome.CREATED
+    }
+    val oldTelephoneAddress = contactDetails.emailAddress?.value
+    if (oldTelephoneAddress != newTelephoneAddress) {
+      updateTelephoneAddress(newTelephoneAddress, contactDetails, oldTelephoneAddress)
+    }
     return SetOutcome.UPDATED
   }
 
+  private fun updateTelephoneAddress(
+    newTelephoneAddress: String,
+    contactDetails: ContactDetails,
+    oldTelephoneAddress: String?,
+  ) {
+    val persistedTelephoneAddress = createOrGetTelephoneAddress(newTelephoneAddress)
+
+    persistedTelephoneAddress.contactDetails.add(contactDetails)
+    contactDetails.telephoneAddress = persistedTelephoneAddress
+    contactDetailsRepository.saveAndFlush(contactDetails)
+
+    oldTelephoneAddress?.let {
+      if (contactDetailsRepository.isTelephoneAddressOrphaned(oldTelephoneAddress)) {
+        telephoneAddressRepository.delete(oldTelephoneAddress)
+      }
+    }
+  }
+
   @Transactional
-  @PreAuthorize(CLIENT_CAN_MAINTAIN_EMAIL_ADDRESSES)
+  @PreAuthorize(CLIENT_CAN_MAINTAIN_ADDRESSES)
   fun deleteEmailAddress(prisonId: String, departmentType: DepartmentType, throwNotFound: Boolean = false) {
     val contactDetails = contactDetailsRepository.getByPrisonIdAndType(prisonId, departmentType)
-
     contactDetails?.let {
```

Comment:
minor comment - 201 and 202 can be combined, same below.

Reply:
I will leave it I think it's more readable really..